### PR TITLE
Group details

### PR DESCRIPTION
**Pull request by edewit**
## Motivation
Added the group details to the group section

## Verification Steps

Go to the groups section create a couple of groups click on the group see the new group details

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented
- [x] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

File: src/realm-roles/RealmRolesSection.css [hunk 1]
```diff
@@ -1,30 +1,4 @@
...
-  margin-left: calc(var(--pf-global--spacer--md) * -1);
-}
-
-.pf-c-button.kc-role-attributes__minus-icon {
-  /* shift the button left to adjust for table cell padding */
-  margin-left: calc(var(--pf-global--spacer--md) * -1);
-  color: var(--pf-c-button--m-plain--Color);
-}
-
-.kc-add-role-button {
```
Comment (jenny-s51) [inline src/realm-roles/RealmRolesSection.css:18]: Looks like we need this styling here

Screen Shot 2021-03-09 at 3 05 18 PM

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
Subgroup functionality is looking good @edewit !

There seems to be a small bug with the `reload()` function in the attributes section...

To reproduce:

1) add a user attribute
2) click 'save'
3) click 'reload'
4) the user attribute field disappears even though the new value should be posted to the server, reload should not remove the field in this case
5) try the same thing in the realm roles section -> when you save an attribute and click 'reload' it does not remove the field for the most recently saved attribute (which is the expected behavior)

File: src/realm-roles/RealmRolesSection.css (hunk 1 above)
Comment (edewit) [reply src/realm-roles/RealmRolesSection.css:18]: Oops seems like I moved one of the styles from here to the new component. But I don't think we even need it. I fixed it by wrapping the controls in `<ToolbarItem>`s

File: src/groups/Members.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  return (
+    <KeycloakDataTable
+      key={key}
+      loader={loader}
+      ariaLabelKey="groups:members"
+      isPaginated
+      toolbarItem={
+        <>
+          <ToolbarItem>
+            <Button data-testid="addMember" variant="primary">
```
Comment (mfrances17) [inline src/groups/Members.tsx:90]: intentionally skipped and will be handled separately?

File: src/groups/GroupTable.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        }
+        actions={[
+          {
+            title: t("moveTo"),
+            onRowClick: () => console.log("TO DO: Add move to functionality"),
+          },
+          {
+            title: t("common:delete"),
+            onRowClick: async (group: GroupRepresentation) => {
+              return deleteGroup(group);
```
Comment (mfrances17) [inline src/groups/GroupTable.tsx:169]: we should provide a confirmation modal on delete, as we do in other parts of the UI

**Review by mfrances17 — CHANGES_REQUESTED** [review]
Found a couple of things. Also, Action > Delete Group and the Add Member button are not working, but that functionality might not be considered part of this PR, if so, ignore.

File: src/groups/Members.tsx (hunk 2 above)
Comment (edewit) [reply src/groups/Members.tsx:90]: yeah this PR was already getting big so I opted to do those in a separate PR

File: src/groups/GroupTable.tsx (hunk 3 above)
Comment (edewit) [reply src/groups/GroupTable.tsx:169]: Right this was already there I'll create an issue for that #430

**Review by mfrances17 — APPROVED** [review]

File: src/groups/GroupAttributes.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  };
+
+  return (
+    <AttributesForm
+      form={form}
+      save={save}
+      array={{ fields, append, remove }}
+      reset={() =>
+        form.reset({
+          attributes: convertAttributes(),
```
Comment (jenny-s51) [inline src/groups/GroupAttributes.tsx:60]:
```suggestion
          attributes: attributes,
```

File: src/groups/GroupAttributes.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import { useTranslation } from "react-i18next";
+import { useFieldArray, useForm } from "react-hook-form";
+import { AlertVariant } from "@patternfly/react-core";
+
+import { useAlerts } from "../components/alert/Alerts";
+import {
+  arrayToAttributes,
+  AttributeForm,
+  AttributesForm,
+  attributesToArray,
```
Comment (jenny-s51) [inline src/groups/GroupAttributes.tsx:11]:
```suggestion
  attributesToArray,
  KeyValueType,
```

File: src/groups/GroupAttributes.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  const { addAlert } = useAlerts();
+  const form = useForm<AttributeForm>({ mode: "onChange" });
+  const { fields, append, remove } = useFieldArray({
+    control: form.control,
+    name: "attributes",
+  });
+
+  const id = getLastId(location.pathname);
+  const { currentGroup, subGroups } = useSubGroups();
+
```
Comment (jenny-s51) [inline src/groups/GroupAttributes.tsx:30]:
```suggestion
  const [ attributes, setAttributes ] = useState<KeyValueType[]>([]);

```

File: src/groups/GroupAttributes.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  const convertAttributes = (attr?: Record<string, any>) => {
+    const attributes = attributesToArray(attr || currentGroup().attributes!);
```
Comment (jenny-s51) [inline src/groups/GroupAttributes.tsx:32]:
```suggestion
    const attributes = attributesToArray(attr || currentGroup().attributes!);
    setAttributes(attributes);
```

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
reload is not quite working as expected - when I click "reload" after saving a new attribute the field disappears ... after some investigating/testing these suggested changes fixed the reload for me 🙂

**Comment by edewit** [comment]
thanks @jenny-s51 I fixed it a little different then you suggested... the reload was still broken after a save... I fixed that now as well

**Review by jenny-s51 — APPROVED** [review]
LGTM! 👍

**Pull request merged**